### DISCUSSION TITLE
APDU_Timeout query

### DISCUSSION THREAD

**Discussion by vishalbacnet** [bacnet-stack/bacnet-stack#695] [Q&A]
![Untitled](https://github.com/user-attachments/assets/703f5a2e-2833-410d-a24a-2bc4465c7c4c)

I am using 5 years old stack atleast, could be much older it shows in yabe 1.9.0 (i am not sure).

when i am subscribing COV and getting Confirmed_COV_Notifications(i am not sending Ack)
then APDU_retry happen starts after 2 secondsm while in yabe i have configured as APDU_Timeput as 6000.
However when retry starts it retried after 3 seconds only.

As of now our device in the BTL lab and this is the last issue whare i request your support.

how can i make APDU_Timeout consistent and under which file i should try to fix it.

if you can suggest i can replace some files from latest stack and fix it, would be great.

**Comment by skarg** [comment]
The cov_send_request() function calls tsm_set_confirmed_unsegmented_transaction() for confirmed COV, which regulates the retries.
TSM uses tsm_timer_milliseconds(uint16_t milliseconds) to check for APDU retries using the number of milliseconds since the last calling, and references apdu_retries() and apdu_timeout() functions for their values, whatever they are set to.
